compiler/rustc_metadata/src/rmeta/encoder.rs

```diff
@@ -1393,7 +1392,13 @@ impl<'a, 'tcx> EncodeContext<'a, 'tcx> {
             if should_encode_fn_sig(def_kind) {
                 record!(self.tables.fn_sig[def_id] <- tcx.fn_sig(def_id));
             }
-            if should_encode_generics(def_kind) {
+            // FIXME: Some anonymous constants produced by `#[rustc_legacy_const_generics]`
+            // do not have corresponding HIR nodes, so some queries usually making sense for
+            // anonymous constants will not work on them and panic. It's not clear whether it
+            // can cause any observable issues or not.
+            let anon_const_without_hir = def_kind == DefKind::AnonConst
```

Comment:
We could just move the def-id encode up to the top then early continue?

Reply:
I suspect that the skipped encoding for a subset of anon consts is a source of bugs that we do not observe only because uses of `#[rustc_legacy_const_generics]` are very rare.
So we should rather try to encode `type_of` and `generics_of` (the only data that is skipped now), than to go back to skipping everything.

Reply:
Makes sense. Can you add a FIXME for those HIR-less queries?

Reply:
There is a FIXME, in the comment above.